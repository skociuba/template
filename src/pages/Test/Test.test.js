import {render} from '@testing-library/react';

import Test from './Test';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

const defaultProps = {
  fetchTestData: jest.fn(),
  testDataSelector: [],
  testExampleSelector: 'hello',
  testLoadingSelector: false,
};

describe('<Test/>', () => {
  it('render without crashing', () => {
    const {container, getByTestId} = render(
      <Test
        fetchTestData={defaultProps.fetchTestData}
        testDataSelector={defaultProps.testDataSelector}
        testExampleSelector={defaultProps.testExampleSelector}
        testLoadingSelector={defaultProps.testLoadingSelector}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByTestId('test-container')).toBeInTheDocument();
  });
});
