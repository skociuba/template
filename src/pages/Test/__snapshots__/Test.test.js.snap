// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Test/> render without crashing 1`] = `
<div
  data-testid="test-container"
>
  <div
    data-testid="header"
  />
  <br />
  <button
    data-testid="button"
  >
    go to main page
  </button>
</div>
`;

exports[`<Test/> render without crashing with testDataSelector 1`] = `
<div
  data-testid="test-container"
>
  <div
    data-testid="header"
  />
  <br />
  <button
    data-testid="button"
  >
    go to main page
  </button>
</div>
`;

exports[`<Test/> render without crashing with testExampleSelector 1`] = `
<div
  data-testid="test-container"
>
  <div
    data-testid="header"
  />
  <br />
  <button
    data-testid="button"
  >
    go to main page
  </button>
</div>
`;

exports[`<Test/> render without crashing with testLoadingSelector 1`] = `
<div
  data-testid="test-container"
>
  <div
    data-testid="header"
  />
  <br />
  <button
    data-testid="button"
  >
    go to main page
  </button>
</div>
`;
